querygen/querygen.go

```diff
@@ -57,3 +57,32 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenBuildTaskRunCompletedQuery returns a Splunk query for generating Segment events
+// when the clair-scan task completes.
+func GenBuildTaskRunCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="taskruns" `+
+			`"objectRef.subresource"="status" `+
+			`"requestObject.metadata.labels.tekton.dev/pipelineTask"="clair-scan" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath path=responseObject.status.conditions{} output=conditions `+
+			`| mvexpand conditions `+
+			`| spath input=conditions `+
+			`| where type="Succeeded" AND reason="Succeeded" AND status="True" `+
+			`| spath path=responseObject.status.taskResults{} output=taskResults `+
+			`| mvexpand taskResults `+
+			`| where name="CLAIR_SCAN_RESULT" `+
```

Comment:
@amisstea Now it should show `45 events` for the last `24 hours` of data.

Reply:
do we really need 8 different splunk commands here? could we simplify things down somehow?

is there some other way to check for some value in some array within the event record other then blowing it up into multiple records via `mvexpend`?

also I commented elsewhere already that:

1.  I'd rather see `search` being used instead of `where` for consistency
2. It rather see `eval` being used instead of `spath` because it makes the result fields being defined explicit.

Reply:
FWIW, while working on the build pipelinerun started events query I didn't find an alternative to `mvexpand` for working with nested arrays but perhaps I didn't look closely enough. @Kousalya1998 please be on the lookout for a better alternative, if possible.

As for using `eval` over `spath` generally, good idea. Some of these nested field names are too generic (e.g. `name`, `value`, etc.). But I think we may need one `spath` command to parse the JSON string value from the `value` field of the task result. This is currently omitted and probably part of the reason this query doesn't return proper results yet.